### ISSUE TITLE
XTD Editor plugin produces an exception when you don't return an object or array

### ISSUE THREAD

**Issue by yvesh** [joomla/joomla-cms#9316]
#### Steps to reproduce the issue

Create a simple Editor XTD Plugin and call a page where an editor is displayed (like com_content edit)

```php
class PlgButtonSample extends JPlugin
{
       public function onDisplay($name, $asset, $author)
	{
           // Can be also a string or boolean, it always breaks the jform
            return;
        }
}

```

#### Expected result

Nothing should be displayed 

#### Actual result

An exception is thrown: (on getInput() where the plugins are loaded)

Call to a member function get() on null 

#### System information (as much as possible)

Staging, php7

#### Additional comments

@wilsonge This is a regression and 3.5 blocker, as it is going to break any old editor plugins which does not return an array or object..

Probably related to https://github.com/joomla/joomla-cms/commit/b50885b200e526eaa4b49e773d8d598387183416

**Comment by infograf768** [comment]
@nonumber 
Please look

**Comment by nonumber** [comment]
This should fix it: #9317

**Comment by infograf768** [comment]
@yvesh 

Please test #9317

**Comment by yvesh** [comment]
@infograf768 
Done, #9317 fixes it

**Comment by brianteeman** [comment]
Closed as #9317 has been merged
